Start using Willow and new image operations

kaedroho opened the pull request:
PR #777 adds Willow support to wagtailimages and PR #780 adds new image operations.

This PR replaces code throughout wagtailimages to start using these new features.

Here's what I've changed:
 - Renditions
 - URL generator (validation of filter spec)
 - Frontend serve view
 - Feature detection

timheap commented on wagtail/wagtailimages/views/frontend.py:
```diff
@@ -14,7 +14,12 @@ def serve(request, signature, image_id, filter_spec):
-    except Filter.InvalidFilterSpecError:
+        # This would raise an exception if the filter spec is invalid
+        fil.operations
+    except:
```
Plain `except:` statements scare me. They have a nasty habit of silencing otherwise interesting and important errors. If an error was thrown thanks to a bug in some code, coupled with the 'Invalid filter spec' error below, will lead to some confusion in the future for some poor developer trying to write a new filter operation. Perhaps catching some relevant errors in the filter spec parsing code, such as:

```python
@cached_property
def operations(self):
    for op_spec in self.spec.split():
        op_spec_parts = op_spec.split('-')
        op_class = self._registered_operations[op_spec_parts[0]]
        try:
            operations.append(op_class(*op_spec_parts))
        except TypeError as e:
            raise InvalidFilterSpecException(e)
    return operations
```

`TypeError`s are raised when a function or constructor is called with the wrong number of arguments, which is the only error that could be caused by this code that is a user mistake. These errors are wrapped in a `InvalidFilterSpecException`. Filters themselves can then raise `InvalidFilterSpecException`s again if the arguments given are bad. The `except:` above could be reverted to a `except InvalidFilterSpecException` to catch all errors with filter specs.

This also allows for the `Filter.is_valid()` function to be kept, which is a useful method as [the change in the next file](https://github.com/torchbox/wagtail/pull/781/files#diff-779c10cbb66261440a434bfaf08e63f1R168) demonstrates.

Any other errors are actual errors that are useful for developers to know about, and should not be swallowed by a bogus "Invalid filter spec" error message.

I realise that the `operation` method is defined in pull request #780, so this change should be made in that one instead.

timheap commented:
Except for my dislike of bare except statements, this code is good.

kaedroho commented:
I've rebased this branch on #780 and updated commits https://github.com/kaedroho/wagtail/commit/e505c08677d5d073ee989cf5ac5f1a074a6a6247 and https://github.com/kaedroho/wagtail/commit/91de79ddc3d2a03014da6ab4d325b8882b7e284b to use the new ``InvalidFilterSpecError`` exception.

timheap commented:
Excellent, these changes look good now.
